### DISCUSSION TITLE
[firebase_database] Why isn't DataSnapshot.value a Map or dynamic in version: ^9.0.0?

### DISCUSSION THREAD

**Discussion by Michael-Eliosov** [firebase/flutterfire#7818] [Q&A] [answered]
Before, I would access a value like DataSnapshot.value[key], now I have to do DataSnapshot.child(key).value or assign DataSnapshot.value to a variable and then variable[key].

Please tell me I'm missing something

**Comment by rrousselGit** [comment]
`value` can be multiple things. It _may_ be a Map, but can also be a String or number, ...

**Comment by Michael-Eliosov** [comment]
Then why doesn't Map a = DataSnapshot.value raise no error?

**Comment by TheHypnoo** [comment, marked as answer]
@Michael-Eliosov 
Now with the new version ^ 9. does the value return an "Object?" that depending on what you return you can treat it in different ways, such as a Map or primitive values ​​(String, int ..)

In this case the .value does not return the ordered values

**Reply by Michael-Eliosov** [reply]
> @Michael-Eliosov Now with the new version ^ 9. does the value return an "Object?" that depending on what you return you can treat it in different ways, such as a Map or primitive values ​​(String, int ..)

My problem is precisely that I cannot treat it as a Map (it gives an error that Object has no method [] )

**Reply by TheHypnoo** [reply]
Could you show an example of it?

**Reply by Michael-Eliosov** [reply]
Sure.

```
FirebaseDatabase.instance.ref("Users").once().then((data) {
   return data.snapshot.value["user"];
}
```

Android Studio says:
> The operator '[]' isn't defined for the type 'Object'.

Meanwhile, this works fine:
```
FirebaseDatabase.instance.ref("Users").once().then((data) {
    Map result = data.snapshot.value;
    return result["user"];
  });
```

**Reply by deepak786** [reply]
you have to cast the value to map now.
```
FirebaseDatabase.instance.ref("Users").once().then((data) {
   return (data.snapshot.value as Map)["user"];
}
```

**Reply by Michael-Eliosov** [reply]
Thanks, though that's really dumb
